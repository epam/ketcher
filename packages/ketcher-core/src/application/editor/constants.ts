export const canvasSelector = '#polymer-editor-canvas';
export const drawnStructuresSelector = '.drawn-structures';
