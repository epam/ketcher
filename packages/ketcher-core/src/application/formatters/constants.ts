export const macromoleculesFilesInputFormats = {
  ket: 'chemical/x-indigo-ket',
  mol: 'chemical/x-mdl-molfile',
  seq: {
    rna: 'chemical/x-rna-sequence',
    dna: 'chemical/x-dna-sequence',
    peptide: 'chemical/x-peptide-sequence',
  },
  fasta: {
    rna: 'chemical/x-rna-fasta',
    dna: 'chemical/x-dna-fasta',
    peptide: 'chemical/x-peptide-fasta',
  },
};
