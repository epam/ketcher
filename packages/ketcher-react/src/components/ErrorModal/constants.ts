export const pasteErrorText =
  'This action is unavailable via menu. Please use shortcuts instead:'

export const shortcuts = [
  { hotKey: 'CTRL/Cmd + C', label: 'for paste' },
  { hotKey: 'CTRL/Cmd + X', label: 'for cut' },
  { hotKey: 'CTRL/Cmd + V', label: 'for paste' }
]

export const error = {
  message: 'Error Message',
  close: 'Close'
}
