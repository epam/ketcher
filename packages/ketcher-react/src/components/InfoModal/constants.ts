export const error = {
  message: 'Error Message',
  close: 'Close',
};
