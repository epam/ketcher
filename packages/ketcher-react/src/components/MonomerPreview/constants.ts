export const preview = {
  width: 345,
  height: 345,
  gap: 5,
  topPadding: 16,
  heightForNucleotide: 105,
  widthForBond: 358,
  heightForBond: 268,
} as const;
