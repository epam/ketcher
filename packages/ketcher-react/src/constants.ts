/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

export const ketcherInitEventName = (ketcherId?: string) =>
  ketcherId ? `ketcher-init-${ketcherId}` : 'ketcher-init';

export const MODES = {
  FG: 'fg',
};

export const STRUCT_TYPE = {
  atoms: 'atoms',
  bonds: 'bonds',
};

export const KETCHER_ROOT_NODE_CLASS_NAME = 'Ketcher-root';
export const KETCHER_ROOT_NODE_CSS_SELECTOR = `.${KETCHER_ROOT_NODE_CLASS_NAME}`;

export const EditorClassName = 'Ketcher-polymer-editor-root';
export const KETCHER_MACROMOLECULES_ROOT_NODE_SELECTOR = `.${EditorClassName}`;
export const STRUCT_SERVICE_NO_RENDER_INITIALIZED_EVENT =
  'struct-service-no-render-initialized';
export const STRUCT_SERVICE_INITIALIZED_EVENT = 'struct-service-initialized';
export const ACS_STYLE_DEFAULT_SETTINGS = {
  atomColoring: false,
  font: '30px Arial',
  fontsz: 10,
  fontszUnit: 'pt',
  fontszsub: 10,
  fontszsubUnit: 'pt',
  reactionComponentMarginSize: 1.6,
  reactionComponentMarginSizeUnit: 'pt',
  imageResolution: '600',
  bondLength: 30,
  bondLengthUnit: 'pt',
  bondSpacing: 18,
  bondThickness: 0.6,
  bondThicknessUnit: 'pt',
  stereoBondWidth: 2,
  stereoBondWidthUnit: 'pt',
  hashSpacing: 2.5,
  hashSpacingUnit: 'pt',
};
