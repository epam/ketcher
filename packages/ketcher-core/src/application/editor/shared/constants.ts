/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { IMAGE_KEY, MULTITAIL_ARROW_KEY } from 'domain/constants';

export const SgContexts = {
  Fragment: 'Fragment',
  Multifragment: 'Multifragment',
  Bond: 'Bond',
  Atom: 'Atom',
  Group: 'Group',
};

export const selectionKeys = [
  'atoms',
  'bonds',
  'frags',
  'sgroups',
  'rgroups',
  'rgroupAttachmentPoints',
  'rxnArrows',
  'rxnPluses',
  'simpleObjects',
  'texts',
  IMAGE_KEY,
  MULTITAIL_ARROW_KEY,
] as const;

export const defaultBondThickness = 2;

export enum MonomerGroups {
  SUGARS = 'Sugars',
  BASES = 'Bases',
  PHOSPHATES = 'Phosphates',
  PEPTIDES = 'Amino acids',
  NUCLEOTIDES = 'Nucleotides',
}

export enum MonomerGroupCodes {
  R = 'R',
  A = 'A',
  C = 'C',
  G = 'G',
  T = 'T',
  U = 'U',
  X = 'X',
  P = 'P',
}

export const MonomerCodeToGroup: Record<MonomerGroupCodes, MonomerGroups> = {
  R: MonomerGroups.SUGARS,
  A: MonomerGroups.BASES,
  C: MonomerGroups.BASES,
  G: MonomerGroups.BASES,
  T: MonomerGroups.BASES,
  U: MonomerGroups.BASES,
  X: MonomerGroups.BASES,
  P: MonomerGroups.PHOSPHATES,
} as const;
