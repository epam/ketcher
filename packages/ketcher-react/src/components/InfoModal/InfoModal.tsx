/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { error } from './constants';

import styles from './InfoModal.module.less';

interface InfoModalProps {
  message: string;
  close: () => void;
}

const InfoModal = ({ message, close }: InfoModalProps): JSX.Element => {
  return (
    <div className={styles.infoModalOverlay}>
      <div
        className={styles.infoModalWindow}
        role="dialog"
        aria-modal="true"
        aria-labelledby="dialog-label"
        aria-describedby="dialog-content"
      >
        <header id="dialog-label">{error.message}</header>
        <div className={styles.infoModalBody} id="dialog-content">
          {message}
        </div>
        <footer>
          <button className={styles.ok} onClick={close}>
            {error.close}
          </button>
        </footer>
      </div>
    </div>
  );
};

export default InfoModal;
