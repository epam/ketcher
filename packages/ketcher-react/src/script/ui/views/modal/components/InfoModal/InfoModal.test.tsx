import { screen } from '@testing-library/react';
import InfoModal from './InfoModal';
import { shortcut } from './constants';
import { renderWithMockStore } from './mockStore';

describe('InfoModal should be rendered correctly', () => {
  it('should render default error message for Cut and Copy actions', () => {
    const props = 'Cut';
    const defaultErrorText = `This action is unavailable via menu. Instead, use shortcut to ${props}.`;
    const view = renderWithMockStore(<InfoModal message={props} />);

    expect(view).toMatchSnapshot();
    expect(screen.getByText(defaultErrorText)).toBeInTheDocument();
  });

  it('should render Paste shortcut message if Paste message dispatched to props', () => {
    const props = 'Paste';
    const view = renderWithMockStore(<InfoModal message={props} />);
    expect(view).toMatchSnapshot();
    expect(screen.getByText(shortcut.hotKey)).toBeInTheDocument();
  });
});
