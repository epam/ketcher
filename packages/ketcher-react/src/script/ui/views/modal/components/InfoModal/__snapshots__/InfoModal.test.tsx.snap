// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal should be rendered correctly should render Paste shortcut message if Paste message dispatched to props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dialog infoModal"
        role="dialog"
        tabindex="-1"
      >
        <header
          class="header"
        >
          <div>
            Error Message
          </div>
          <div
            class="btnContainer"
          >
            <button
              class="buttonTop"
            >
              <icon-close.svg
                classname="closeButton"
              />
            </button>
          </div>
        </header>
        <div
          class="body withMargin"
        >
          <div>
            <div
              class="pasteErrorModalBody"
            >
              <div>
                Your browser doesn't allow pasting clipboard content via button. Please use shortcut instead.
              </div>
              <div
                class="shortcutsBlock"
              >
                <div
                  class="shortcuts"
                >
                  <div
                    class="shortcut"
                    data-testid="infoModal-shortcut-for-paste"
                  >
                    CTRL/Cmd + V
                  </div>
                  <div>
                    for paste
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <footer
          class="footer"
        >
          <button
            class="ok"
          >
            Close
          </button>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dialog infoModal"
      role="dialog"
      tabindex="-1"
    >
      <header
        class="header"
      >
        <div>
          Error Message
        </div>
        <div
          class="btnContainer"
        >
          <button
            class="buttonTop"
          >
            <icon-close.svg
              classname="closeButton"
            />
          </button>
        </div>
      </header>
      <div
        class="body withMargin"
      >
        <div>
          <div
            class="pasteErrorModalBody"
          >
            <div>
              Your browser doesn't allow pasting clipboard content via button. Please use shortcut instead.
            </div>
            <div
              class="shortcutsBlock"
            >
              <div
                class="shortcuts"
              >
                <div
                  class="shortcut"
                  data-testid="infoModal-shortcut-for-paste"
                >
                  CTRL/Cmd + V
                </div>
                <div>
                  for paste
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <button
          class="ok"
        >
          Close
        </button>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`InfoModal should be rendered correctly should render default error message for Cut and Copy actions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dialog infoModal"
        role="dialog"
        tabindex="-1"
      >
        <header
          class="header"
        >
          <div>
            Error Message
          </div>
          <div
            class="btnContainer"
          >
            <button
              class="buttonTop"
            >
              <icon-close.svg
                classname="closeButton"
              />
            </button>
          </div>
        </header>
        <div
          class="body withMargin"
        >
          <div>
            This action is unavailable via menu. Instead, use shortcut to Cut.
          </div>
        </div>
        <footer
          class="footer"
        >
          <button
            class="ok"
          >
            Close
          </button>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dialog infoModal"
      role="dialog"
      tabindex="-1"
    >
      <header
        class="header"
      >
        <div>
          Error Message
        </div>
        <div
          class="btnContainer"
        >
          <button
            class="buttonTop"
          >
            <icon-close.svg
              classname="closeButton"
            />
          </button>
        </div>
      </header>
      <div
        class="body withMargin"
      >
        <div>
          This action is unavailable via menu. Instead, use shortcut to Cut.
        </div>
      </div>
      <footer
        class="footer"
      >
        <button
          class="ok"
        >
          Close
        </button>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
